src/app/components/authorization/authorization.tsx

```diff
@@ -0,0 +1,153 @@
+import React, { useState, useRef } from 'react'
+import "./authorization.css"
+import { createUserWithEmailAndPassword, sendEmailVerification, signInWithEmailAndPassword } from "firebase/auth"
+import { auth } from "../../../firebase-config"
+import Button from '@mui/material/Button'
+import { Box, TextField, Typography } from '@mui/material'
+import { Link, useNavigate } from 'react-router-dom'
+import bigLogo from '../../assets/big-logo.jpg'
+import ErrorIcon from '@mui/icons-material/Error'
+
+function Authorization(): JSX.Element {
+  const [registrationEmail, setRegistrationEmail] = useState("");
+  const [registrationPassword, setRegistrationPassword] = useState("");
+
+  const [loginEmail, setLoginEmail] = useState("");
+  const [loginPassword, setLoginPassword] = useState("");
+
+  const [authorizationHendler, setAuthorizationHenadler] = useState<string>("register");
+
+  const [authorizationError, setAuthorizationError] = useState<string>("");
+
+  const registerHendlerRef = useRef<any>(null);
+  const loginHendlerRef = useRef<any>(null);
+
+  const registerEmailRef = useRef<any>(null);
+  const registerPasswordRef = useRef<any>(null);
+  const loginEmailRef = useRef<any>(null);
+  const loginPasswordRef = useRef<any>(null);
+
+  const navigate = useNavigate();
+  const register = async () => {
+    try {
```

Comment:
Put these functions outside of a component as js will recreate them from scratch at each rerender.
Actually, these functions have to be in the redux or some state management service, but not your component.